rust/core-lib/src/session.rs

```diff
@@ -0,0 +1,54 @@
+use xi_rope::rope::RopeInfo;
+use xi_rope::delta::Delta;
+use tabs::ViewId;
+use serde_json::{self, Value};
+
+use std::time::{Duration, Instant};
+use std::collections::HashMap;
+use std::fs::OpenOptions;
+use std::io::Write;
+
+// NOTE: increase delay to notice the effect
+pub(crate) const SESSION_SAVE_DELAY: Duration = Duration::from_millis(1000);
+
+// TODO: is this OK?
+pub(crate) const SESSION_SAVE_MASK: usize = 1 << 26;
```

Comment:
maybe we should declare these all in one place to make it easier to avoid collisions?